algorithms/benches/snark/varuna.rs

```diff
@@ -66,13 +67,19 @@ fn snark_prove(c: &mut Criterion) {
 
         let max_degree = AHPForR1CS::<Fr, VarunaHidingMode>::max_degree(1000, 1000, 1000).unwrap();
         let universal_srs = VarunaInst::universal_setup(max_degree).unwrap();
-        let universal_prover = &universal_srs.to_universal_prover().unwrap();
         let fs_parameters = FS::sample_parameters();
+        let hiding_bound = AHPForR1CS::<Fr, VarunaHidingMode>::zk_bound().unwrap_or(0);
 
         let (circuit, _) = TestCircuit::gen_rand(mul_depth, num_constraints, num_variables, rng);
 
-        let params = VarunaInst::circuit_setup(&universal_srs, &circuit).unwrap();
-        b.iter(|| VarunaInst::prove(universal_prover, &fs_parameters, &params.0, &circuit, rng).unwrap())
+        let (pk, _) = VarunaInst::circuit_setup(&universal_srs, &circuit).unwrap();
+        let max_degree = pk.circuit.max_degree();
+        let max_domain_size = pk.circuit.max_domain_size();
+        let coefficient_support = pk.circuit.index_info.get_degree_bounds::<Fr>();
+        let universal_prover = &universal_srs
```

Comment:
Maybe this API should take as input a bunch of `CircuitInfo`s so that it can perform the above calculation internally?

Reply:
That way we don't have to peer inside the contents of the `pk` to construct `universal_prover`.

Reply:
If not a `CircuitInfo`, maybe a `DegreeAndDomainInfo` or something like that.

Reply:
> Maybe this API should take as input a bunch of CircuitInfos so that it can perform the above calculation internally?

Just so we're on the same page, this is a slighly different recommendation than you give below. We can pass either `pk`, `pk.circuit`, `pk.circuit.index_info` or `pk.circuit.index_info.degree_info()` into universal prover. I'm doing the latter now, which seems cleaner than passing the entire `pk`.